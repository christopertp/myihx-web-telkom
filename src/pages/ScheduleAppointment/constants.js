export const ASSURANCE_FETCHED = 'ScheduleAppointment/assurance-fetched';
export const DATE_FETCHED = 'ScheduleAppointment/date-fetched';
export const FAILED = 'ScheduleAppointment/failed';
export const INFO_FETCHED = 'ScheduleAppointment/info-fetched';
export const ISSUE_ID_FETCHED = 'ScheduleAppointment/issue-id-fetched';
export const LOADING = 'ScheduleAppointment/loading';
