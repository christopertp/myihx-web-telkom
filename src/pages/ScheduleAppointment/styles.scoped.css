.address {
  margin-bottom: 1.5rem;
}

.address > p {
  color: #333333;
  line-height: 1.4;
  margin-top: 0.5rem;
}

.address > span {
  display: block;
  height: 1rem;
  margin-top: 0.5rem;
  width: 70%;
}

.buttons { display: flex; }

.buttons > *:first-child {
  margin-right: 1rem;
  flex: 1;
}

.buttons > *:last-child { flex: 1; }

.date {
  background-color: #f2f2f2;
  margin: 0 -1.5rem 1rem;
  padding: 1rem 1.5rem;
}

.error {
  color: #ee4c24;
  display: block;
  font-size: 1rem;
  margin-bottom: 1rem;
}

.issue {
  margin-bottom: 1.5rem;
}

.issue > div {
  color: #333333;
  line-height: 1.4;
  margin-top: 0.5rem;
}

.issue > div > span {
  display: block;
  height: 1rem;
  width: 60%;
}

.modal-404 > img { height: 6.25rem; }
.modal-404 > h3 { margin-bottom: 1rem; }

.modal-timeout > h3 {
  color: #191a1a;
  font-size: 1.5rem;
  text-align: center;
}

.modal-timeout > p {
  color: #191a1a;
  margin: 0.5rem 0 2.25rem;
  text-align: center;
}

.modal-timeout > button { width: 100%; }

.success-date {
  background-color: #f2f2f2;
  margin-top: 1.25rem;
  padding: 1rem;
  text-align: center;
}

.success-date > h3 {
  font-size: 1.25rem;
  margin-bottom: 0.25rem;
}

.success-date > p {
  color: #66686a;
}

.success-date + a {
  margin: 1.25rem 0 0;
}

.timer {
  position: absolute;
  right: 1rem;
  top: 1rem;
}

@media only screen and (max-width: 48em) {
  .address { margin-bottom: 1.625rem; }
  .address > p { font-size: 0.9rem; }

  .issue { margin-bottom: 2rem; }
  .issue > div { font-size: 0.9rem; }

  .success-date > h3 { font-size: 1rem; }
  .success-date > p { font-size: 0.75rem; }
  .success-date + a { margin: auto 0 1.5rem; }
}
