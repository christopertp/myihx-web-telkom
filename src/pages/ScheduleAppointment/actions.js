import { push } from 'react-router-redux';
import * as assurance from '../../services/assurance';
import * as order from '../../services/order';
import { ASSURANCE_FETCHED, DATE_FETCHED, FAILED, INFO_FETCHED, ISSUE_ID_FETCHED, LOADING } from './constants';

export function fetchPost(payload, type) {
  return async dispatch => {
    dispatch(loadingAction(true, 'submit'));
    try {
      if (type === 'service') {
        const mapPayload = { ... payload, schedule: payload.timeSlot };
        delete mapPayload.timeSlot;
        const { data } = await assurance.postSchedule(mapPayload);
        dispatch(issueIdFetched(data.issueId));
      } else {
        await order.postSchedule(payload);
        dispatch(failedAction('', 'submit'));
      }
      dispatch(push(`/schedule-${type}/success`));
    } catch (err) {
      dispatch(failedAction(err.message, 'submit'));
    }
  };
}

export function fetchDate(type) {
  return async dispatch => {
    dispatch(loadingAction(true, 'date'));
    try {
      if (type === 'service') {
        const { data } = await assurance.getSchedule();
        const { address, issueSummary, schedule } = data;
        dispatch(assuranceFetchedAction(address.description, schedule, issueSummary.id));
      } else {
        const { data } = await order.getSchedule();
        dispatch(dateFetchedAction(data.availability));
      }
    } catch (err) {
      dispatch(failedAction(err.code === 404 ? '404' : 'error', 'date'));
    }
  };
}

export function fetchInstallationInfo() {
  return async dispatch => {
    dispatch(loadingAction(true, 'info'));
    try {
      const { data } = await order.getSubscriptions();
      const { product, installationAddress } = data;
      const address = installationAddress.description;
      const dataProduct = {
        description: product.flag,
        id: product.productId,
        mbps: product.internetPackage.speed,
        name: product.packageName,
        price: product.price,
      };
      dispatch(infoFetchedAction(address , dataProduct));
    } catch (err) {
      dispatch(infoFetchedAction('', ''));
    }
  };
}

function assuranceFetchedAction(address, date, issue) {
  return { type: ASSURANCE_FETCHED, address, date, issue };
}

function dateFetchedAction(date) {
  return { type: DATE_FETCHED, date };
}

function failedAction(message, key) {
  return { type: FAILED, message, key };
}

function infoFetchedAction(address, product) {
  return { type: INFO_FETCHED, address, product };
}

function issueIdFetched(issueId) {
  return { type: ISSUE_ID_FETCHED, issueId };
}

function loadingAction(isLoading, key) {
  return { type: LOADING, isLoading, key };
}
