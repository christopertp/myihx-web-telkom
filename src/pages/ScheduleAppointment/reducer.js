import { ASSURANCE_FETCHED, DATE_FETCHED, FAILED, INFO_FETCHED, ISSUE_ID_FETCHED, LOADING } from './constants';

const initialState = {
  isLoading: {
    d: true,
    i: true,
    s: false,
  },
  address: '',
  date: [],
  issue: '',
  issueId: '',
  message: '',
  product: '',
};

export default function reducer(state = initialState, action = {}) {
  const { type, address, isLoading, date, message, issue, issueId, product, key } = action;

  switch (type) {
    case ASSURANCE_FETCHED:
      return {
        ...state,
        address,
        date: setAssuranceDate(date),
        issue,
        isLoading: { ...state.isLoading, d: false },
      };
    case DATE_FETCHED:
      return {
        ...state,
        date: setDate(date),
        isLoading: { ...state.isLoading, d: false },
      };
    case FAILED:
      return {
        ...state,
        message,
        isLoading: { ...state.isLoading, [key.charAt(0)]: false },
      };
    case INFO_FETCHED:
      return {
        ...state,
        address,
        isLoading: { ...state.isLoading, i: false },
        product,
      };
    case ISSUE_ID_FETCHED:
      return {
        ...state,
        isLoading: { ...state.isLoading, s: false },
        issueId,
        message: '',
      };
    case LOADING:
      return {
        ...state,
        isLoading: { ...state.isLoading, [key.charAt(0)]: isLoading },
      };
    default:
      return state;
  }
}

export function setAssuranceDate(date) {
  const initial = { dates: [], data: [], };
  return date.reduce((acc, i) => {
    const { dates, data } = acc;
    const clean = i.date.split('-').reverse().join('-');
    const idx = dates.indexOf(clean);

    const newArrival = {
      name: i.slotId,
      availability: i.availability,
      slot: i.timeSlot.slot,
      time: i.timeSlot.time,
      data: i,
    };
    const names = ['early-morning', 'morning', 'noon', 'evening'];
    const idxName = names.indexOf(newArrival.name);

    if (idx === -1) {
      const arrival = ['', '', '', ''];
      arrival[idxName] = newArrival;
      dates.push(clean);
      data.push({ date: clean, arrival });
    } else {
      data[idx].arrival[idxName] = newArrival;
    }

    return { dates, data };
  }, initial).data;
}

export function setDate(date) {
  const count = val => (
    parseFloat(val.split(':')[0]) + (parseFloat(val.split(':')[1])/100)
  );

  const pushNewDate = (acc, newArrival, i) => {
    const { dates, data } = acc;
    const arrival = newArrival.name === 'Morning' ? [newArrival, ''] : ['' ,newArrival];
    const newData = { date: i.date, arrival, };
    dates.push(i.date);
    data.push(newData);
    return { dates, data };
  };

  const pushMorning = (arrival, newArrival) => {
    const isNew = arrival[0] ? count(newArrival.time) < count(arrival[0].time) : true;
    return isNew ? [newArrival, arrival[1]] : arrival;
  };

  const pushAfternoon = (arrival, newArrival) => {
    const isNew = arrival[1] ? count(newArrival.time) < count(arrival[1].time) : true;
    return isNew ? [arrival[0], newArrival] : arrival;
  };

  const initial = { dates: [], data: [], };
  return date.reduce((acc, i) => {
    const { dates, data } = acc;
    const idx = dates.indexOf(i.date);
    const newArrival = {
      name: i.data.partsOfDay,
      availability: i.data.availability,
      slot: i.data.timeSlot.slot,
      time: i.data.timeSlot.time,
      data: i.data,
    };

    if (idx === -1) {
      return pushNewDate(acc, newArrival, i);
    }

    const { arrival } = data[idx];
    const newData = { date: i.date };
    newData.arrival = newArrival.name === 'Morning' ? pushMorning(arrival, newArrival) : pushAfternoon(arrival, newArrival);
    data[idx] = newData;
    return { dates, data };
  }, initial).data;
}
