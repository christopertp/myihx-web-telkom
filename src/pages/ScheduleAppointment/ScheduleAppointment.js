import React, { createContext, useContext, useEffect, useState } from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import { useHistory, useParams } from 'react-router-dom';
import AppointmentDate from '../../components/elements/AppointmentDate';
import Button from '../../components/elements/Button';
import Card from '../../components/elements/Card';
import Heading from '../../components/elements/Heading';
import SectionPackage from '../../components/elements/InstallationPackageInfo';
import Modal from '../../components/elements/Modal';
import Spinner from '../../components/elements/Spinner';
import Timer from '../../components/elements/Timer';
import SectionContact from '../../components/fragments/ScheduleAppointmentContact';
import WhiteCardPlain from '../../components/layouts/WhiteCardPlain';
import WhiteCardSuccessPlain from '../../components/layouts/WhiteCardSuccessPlain';
import useTimer from '../../hooks/useTimer';
import { fetchDate, fetchInstallationInfo, fetchPost } from './actions';
import styles from './styles.scoped.css';

const Context = createContext({});

const TITLE = {
  installation: 'Schedule Installation',
  service: 'Schedule Service',
};

export default function ScheduleAppointment() {
  const history = useHistory();
  const { page, type } = useParams();
  const [modal, setModal] = useState('');
  const [payload, setPayload] = useState({ timeSlot: { date: '' } });

  useEffect(() => {
    if (!type) {
      history.replace('/');
    } else if (page) {
      history.replace(`/schedule-${type}`);
    }
  }, []);

  const contextValue = {
    modal, setModal,
    payload, setPayload,
  };

  return (
    <Context.Provider value={contextValue}>
      {(p => {
        return (p === 'success') ? <Success /> : <Schedule />;
      })(page)}
    </Context.Provider>
  );
}

export function Schedule() {
  const initTime = (10 * 60) - 1;
  const { modal, setModal, payload, setPayload } = useContext(Context);
  const { type } = useParams();
  const { address, date, isLoading, message, product } = useSelector(s => s.scheduleAppointment);
  const { setTimer, timer, time } = useTimer(initTime);
  const dispatch = useDispatch();
  const isInstallation = type === 'installation';
  const subtitle = 'Finish your schedule before the timer runs out to book a technician.';
  const btnDisabled = isLoading.s || !payload.bookingId;

  const payloadReducer = v => setPayload({ ...payload, ...v });

  const onRetry = () => {
    if (modal === 'timeout') {
      setTimer(initTime);
      dispatch(fetchDate(type));
    }
    setModal('');
  };

  const onSubmit = () => {
    dispatch(fetchPost(payload, type));
  };

  useEffect(() => {
    (isInstallation) && dispatch(fetchInstallationInfo());
    dispatch(fetchDate(type));
  }, []);

  useEffect(() => {
    (message === '404') && setModal('404');
  }, [message]);

  useEffect(() => {
    !timer && setModal('timeout');
  }, [timer]);

  return (
    <>
      <WhiteCardPlain back="/" subtitle={subtitle} title={TITLE[type]}>
        <Timer className={styles.timer} time={time} />
        {
          isInstallation
            ? <SectionPackage address={address} isLoading={isLoading.i} product={product} />
            : <IssueInfo />
        }
        <AppointmentDate
          className={styles.date}
          date={date}
          isLoading={isLoading.d}
          mini={!isInstallation}
          setPayload={payloadReducer}
        />
        <SectionContact data={payload.contactSecondary} open={modal === 'contact'} setModal={setModal} setPayload={payloadReducer} />
        {(message && message !== '404') && <small className={styles.error}>{message}</small>}
        <div className={styles.buttons}>
          <Button to="/" variant="bordered">Cancel</Button>
          <Button disabled={btnDisabled} onClick={onSubmit}>
            {isLoading.s ? <Spinner /> : 'Next'}
          </Button>
        </div>
      </WhiteCardPlain>
      <Modals onRetry={onRetry} />
    </>
  );
}

export function IssueInfo() {
  const { address, isLoading, issue } = useSelector(s => s.scheduleAppointment);
  const loading = <span className="loading" />;
  return (
    <>
      <section className={styles.issue}>
        <Heading>Issue Summary</Heading>
        <Card variant="hover">{isLoading.d ? loading : issue}</Card>
      </section>
      <section className={styles.address}>
        <Heading>Address</Heading>
        {isLoading.d ? loading : <p>{address}</p>}
      </section>
    </>
  );
}

export function Modals({ onRetry }) {
  const { modal } = useContext(Context);
  const { type } = useParams();
  return (
    <>
      <Modal className={styles['modal-timeout']} onClose={onRetry} open={modal === 'timeout'}>
        <h3>Session Time Out</h3>
        <p>
          Sorry, your session is expired.
          Retry and try to finish your schedule before the timer expires.
        </p>
        <Button onClick={onRetry}>Retry</Button>
      </Modal>
      <Modal className={styles['modal-404']} onClose={onRetry} open={modal === '404'}>
        <img alt="" src="/assets/grfx_warning.svg" />
        <h3>No {type} to be scheduled</h3>
        <Button to="/">Back to Home</Button>
      </Modal>
    </>
  );
}

Modals.propTypes = {
  onRetry: PropTypes.func.isRequired,
};

export function Success() {
  const { payload } = useContext(Context);
  const { type } = useParams();
  const { issueId } = useSelector(s => s.scheduleAppointment);

  const { date, slot } = payload.timeSlot;
  const title = `All set! Your ${type} is scheduled`;
  const subtitle = type === 'service' ? 'Our technician will visit you on' : 'Your IndiHome will be installed on';
  const cleanDate = type === 'service' ? date.split('-').reverse().join('-') : date;

  return (
    <WhiteCardSuccessPlain icon="success" subtitle={subtitle} title={title}>
      <div className={styles['success-date']}>
        <h3>{moment(cleanDate).format('dddd, D MMMM YYYY')}</h3>
        <p>Estimated arrival time {slot}</p>
      </div>
      {
        type === 'service' ?
          <Button to={`/issues/${issueId}`} variant="white">View Report Progress</Button> :
          <Button to="/">Back to Home</Button>
      }
    </WhiteCardSuccessPlain>
  );
}
