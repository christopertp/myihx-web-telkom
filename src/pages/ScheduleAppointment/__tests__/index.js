import React, { useContext } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useHistory, useParams } from 'react-router-dom';
import ShallowRenderer from 'react-test-renderer/shallow';
import Spinner from '../../../components/elements/Spinner';
import useTimer from '../../../hooks/useTimer';
import ScheduleAppointment, { Schedule, IssueInfo, Modals, Success } from '../ScheduleAppointment';

jest.mock('../../../hooks/useTimer', () => jest.fn(() => ({})));

jest.mock('../actions', () => ({
  fetchDate: jest.fn(),
  fetchInstallationInfo: jest.fn(),
  fetchPost: jest.fn(),
}));

const context = {
  modal: '',
  setModal: jest.fn(),
  setPayload: jest.fn(),
  payload: {
    timeSlot: {
      date: '',
    },
  },
};
useContext.mockImplementation(() => context);

useParams.mockImplementation(() => ({
  page: '',
  type: 'installation',
}));

useSelector.mockImplementation(fn => {
  fn({});
  return { isLoading: '', issueId: 'MIXN-01' };
});

describe('src/pages/ScheduleAppointment', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<ScheduleAppointment />);
    expect(tree).toMatchSnapshot();
  });

  test('render subroute', () => {
    const replace = jest.fn();
    useHistory.mockImplementationOnce(() => ({ replace }));
    useParams.mockImplementationOnce(() => ({ page: 'success', type: 'installation' }));

    const result = ScheduleAppointment();
    expect(replace).toHaveBeenCalledWith('/schedule-installation');
    expect(result.props.children.type).toBe(Success);

    useHistory.mockImplementationOnce(() => ({ replace }));
    useParams.mockImplementationOnce(() => ({ type: '' }));

    ScheduleAppointment();
    expect(replace).toHaveBeenCalledWith('/');
  });

  test('Schedule', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Schedule />);
    expect(tree).toMatchSnapshot();
  });

  test('Schedule error, onRetry, & loading state', () => {
    const setModal = jest.fn();
    useContext.mockImplementationOnce(() => ({ ...context, modal: 'timeout', setModal }));

    useSelector.mockImplementationOnce(() => ({ isLoading: { s: true }, message: 'error' }));

    const setTimer = jest.fn();
    useTimer.mockImplementationOnce(() => ({ setTimer, timer: true }));

    const result = Schedule();
    const whiteCardChildren = result.props.children[0].props.children;
    expect(whiteCardChildren[4].props.children).toBe('error');
    expect(whiteCardChildren[5].props.children[1].props.children.type).toBe(Spinner);

    result.props.children[1].props.onRetry();
    expect(setTimer).toHaveBeenCalled();
    expect(setModal).toHaveBeenCalledWith('');
  });

  test('Schedule modal 404, IssueInfo, & modal onClose', () => {
    const setModal = jest.fn();
    useContext.mockImplementationOnce(() => ({ ...context, setModal }));

    useParams.mockImplementationOnce(() => ({ type: 'service' }));

    useSelector.mockImplementationOnce(() => ({ isLoading: '', message: '404' }));
    const result = Schedule();
    expect(setModal).toHaveBeenCalledWith('404');
    expect(result.props.children[0].props.children[1].type).toBe(IssueInfo);

    result.props.children[1].props.onRetry();
    expect(setModal).toHaveBeenCalledWith('');
  });

  test('Schedule payloadReducer, and onSubmit', () => {
    const setPayload = jest.fn();
    useContext.mockImplementationOnce(() => ({ ...context, payload: {}, setPayload }));

    const dispatch = jest.fn();
    useDispatch.mockImplementationOnce(() => dispatch);

    const result = Schedule();
    const whiteCardChildren = result.props.children[0].props.children;
    whiteCardChildren[2].props.setPayload({ data: 'tes' });
    expect(setPayload).toHaveBeenCalledWith({ data: 'tes' });

    whiteCardChildren[5].props.children[1].props.onClick();
    expect(dispatch).toHaveBeenCalled();
  });

  test('IssueInfo', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<IssueInfo />);
    expect(tree).toMatchSnapshot();
  });

  test('IssueInfo loading state', () => {
    useSelector.mockImplementationOnce(() => ({ isLoading: { d: true } }));
    const result = IssueInfo();
    const [issue, address] = result.props.children;
    expect(issue.props.children[1].props.children.props.className).toBe('loading');
    expect(address.props.children[1].props.className).toBe('loading');
  });

  test('Modals', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Modals onRetry={() => {}} />);
    expect(tree).toMatchSnapshot();
  });

  test('Success', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Success />);
    expect(tree).toMatchSnapshot();
  });

  test('Success service page', () => {
    useParams.mockImplementationOnce(() => ({ type: 'service' }));
    const result = Success();
    expect(result.props.subtitle).toBe('Our technician will visit you on');
    expect(result.props.children[1].props.children).toBe('View Report Progress');
  });
});
