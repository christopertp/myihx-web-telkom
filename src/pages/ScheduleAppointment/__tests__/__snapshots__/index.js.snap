// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/pages/ScheduleAppointment IssueInfo 1`] = `
<React.Fragment>
  <section>
    <Heading
      className=""
    >
      Issue Summary
    </Heading>
    <Card
      className=""
      disabled={false}
      onClick={null}
      to=""
      variant="hover"
    />
  </section>
  <section>
    <Heading
      className=""
    >
      Address
    </Heading>
    <p />
  </section>
</React.Fragment>
`;

exports[`src/pages/ScheduleAppointment Modals 1`] = `
<React.Fragment>
  <Modal
    className=""
    onClose={[Function]}
    open={false}
  >
    <h3>
      Session Time Out
    </h3>
    <p>
      Sorry, your session is expired. Retry and try to finish your schedule before the timer expires.
    </p>
    <Button
      className=""
      disabled={false}
      onClick={[Function]}
      to=""
      type="button"
      variant="red"
    >
      Retry
    </Button>
  </Modal>
  <Modal
    className=""
    onClose={[Function]}
    open={false}
  >
    <img
      alt=""
      src="/assets/grfx_warning.svg"
    />
    <h3>
      No 
      installation
       to be scheduled
    </h3>
    <Button
      className=""
      disabled={false}
      onClick={null}
      to="/"
      type="button"
      variant="red"
    >
      Back to Home
    </Button>
  </Modal>
</React.Fragment>
`;

exports[`src/pages/ScheduleAppointment Schedule 1`] = `
<React.Fragment>
  <WhiteCardPlain
    back="/"
    close=""
    help={false}
    icon=""
    size="large"
    subtitle="Finish your schedule before the timer runs out to book a technician."
    title="Schedule Installation"
  >
    <Timer
      className=""
      time={
        Array [
          "",
          "",
          "",
        ]
      }
    />
    <InstallationPackageInfo
      address=""
      isLoading={false}
      product=""
    />
    <AppointmentDate
      className=""
      date={Array []}
      isLoading={false}
      mini={false}
      setPayload={[Function]}
    />
    <ScheduleAppointmentContact
      data={null}
      open={false}
      setModal={
        [MockFunction] {
          "calls": Array [
            Array [
              "timeout",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      setPayload={[Function]}
    />
    <div>
      <Button
        className=""
        disabled={false}
        onClick={null}
        to="/"
        type="button"
        variant="bordered"
      >
        Cancel
      </Button>
      <Button
        className=""
        disabled={true}
        onClick={[Function]}
        to=""
        type="button"
        variant="red"
      >
        Next
      </Button>
    </div>
  </WhiteCardPlain>
  <Modals
    onRetry={[Function]}
  />
</React.Fragment>
`;

exports[`src/pages/ScheduleAppointment Success 1`] = `
<WhiteCardSuccessPlain
  close=""
  icon="success"
  subtitle="Your IndiHome will be installed on"
  title="All set! Your installation is scheduled"
>
  <div>
    <h3>
      dddd, D MMMM YYYY
    </h3>
    <p>
      Estimated arrival time 
    </p>
  </div>
  <Button
    className=""
    disabled={false}
    onClick={null}
    to="/"
    type="button"
    variant="red"
  >
    Back to Home
  </Button>
</WhiteCardSuccessPlain>
`;

exports[`src/pages/ScheduleAppointment render 1`] = `
<Provider
  value={
    Object {
      "modal": "",
      "payload": Object {
        "timeSlot": Object {
          "date": "",
        },
      },
      "setModal": [MockFunction],
      "setPayload": [MockFunction],
    }
  }
>
  <Schedule />
</Provider>
`;
